pkg/common/cns-lib/volume/manager.go

```diff
@@ -279,10 +279,21 @@ func (m *defaultManager) CreateVolume(ctx context.Context, spec *cnstypes.CnsVol
 		datastoreURL = dsMo.Summary.Url
 	}
 
-	blockBackingDetails, ok := spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+	blockBackingDetails, isBlockBackingDetails := spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+	fileBackingDetails, isFileBackingDetails := spec.BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails)
+
+	var isStaticallyProvisionedBlockVolume bool
+	var isStaticallyProvisionedFileVolume bool
 	// Remove this task from volumeTaskMap in case successful static volume provisioning
 	// as it doesn't result in orphaned volumes
-	if ok && (blockBackingDetails.BackingDiskId != "" || blockBackingDetails.BackingDiskUrlPath != "") {
+	if spec.VolumeType == string(cnstypes.CnsVolumeTypeBlock) && isBlockBackingDetails && (blockBackingDetails.BackingDiskId != "" || blockBackingDetails.BackingDiskUrlPath != "") {
+		isStaticallyProvisionedBlockVolume = true
+	}
+	if spec.VolumeType == string(cnstypes.CnsVolumeTypeFile) && isFileBackingDetails && fileBackingDetails.BackingFileId != "" {
+		isStaticallyProvisionedFileVolume = true
+	}
+
+	if isStaticallyProvisionedBlockVolume || isStaticallyProvisionedFileVolume {
```

Comment:
Can you correct this condition as below?

`if !isStaticallyProvisionedBlockVolume && !isStaticallyProvisionedFileVolume {`

Reply:
if that is the case then why do we need to add this check - `if isStaticallyProvisionedBlockVolume || isStaticallyProvisionedFileVolume {`?

We do not want to retrieve tasks only when the volume is statically provisioned.

Reply:
Extending the static provisioning fix to file volumes

Reply:
ok got it. here we are deleting a task. but can we not place the task in the map, if we determine volume being created is static volume.

Reply:
That's reasonable too! Will move this check above, to avoid storing the tasks in the first place

Reply:
Done! Verified with the latest commit:


```
kubectl logs vsphere-csi-controller-6b8fbfb48c-5c7r7 -c vsphere-syncer -n kube-system | grep "CreateVolume: Volume created successfully"
2020-12-16T19:59:32.430Z	INFO	volume/manager.go:301	CreateVolume: Volume created successfully. VolumeName: "static-pv-name", opId: "98ce3a03", volumeID: "54f1aade-bada-4d93-b0a6-fe6cecc765e9"	{"TraceId": "449e99a2-35e2-45da-9223-93cb60221b54"}
2020-12-16T20:02:26.469Z	INFO	volume/manager.go:301	CreateVolume: Volume created successfully. VolumeName: "static-pv-name", opId: "98ce3a2e", volumeID: "556798ca-5ed2-4da8-8ff8-ca428906bed2"	{"TraceId": "accf79a0-ca02-436d-9d3d-bd9cefeab392"}
```